src/app/Features/3Commas/3Commas.ts

```diff
@@ -235,7 +237,7 @@ const fetchBotPerformanceMetrics = async (profileData: Type_Profile, oDate?: Dat
                 GROUP BY
                     bot_id;`
 
-    let databaseQuery = await window.ThreeCPM.Repository.Database.query(queryString);
+    let databaseQuery: fetchBotPerformanceMetrics[] | [] = await window.ThreeCPM.Repository.Database.query(queryString);
```

Comment:
Just out of curiosity, why do we have those double typings on arrays? I mean the `<T>[] | []`

Reply:
Just because the database query can also return an empty array of nothing is found.

Reply:
Gotcha thanks! Maybe on a future iteration we could detect empty lines that at the `database` stage and either:
- throw an Typed exception (NoResultException)
- return a typed reject promise (kindof the same as above)
- force the type of the empty array

Reply:
Great, turned this into an issue to be fixed!